sdk/src/suins-client.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import type { SuiClient } from '@mysten/sui.js/client';
+
+import {
+	getConfigType,
+	getDomainType,
+	getPricelistConfigType,
+	getRenewalPricelistConfigType,
+	MAINNET_CONFIG,
+	TESTNET_CONFIG,
+} from './constants.js';
+import { isSubName, parsePriceListFromConfig, validateYears } from './helpers.js';
+import type { Constants, NameRecord, SuinsClientConfig, SuinsPriceList } from './types.js';
+import { isValidSuiNSName } from '@mysten/sui.js/utils';
+
+/// The SuinsClient is the main entry point for the Suins SDK.
+/// It allows you to interact with SuiNS.
+export class SuinsClient {
+	#client: SuiClient;
+	constants: Constants = {};
+
+	constructor(config: SuinsClientConfig) {
+		this.#client = config.client;
+		const network = config.network || 'mainnet';
+
+		if (network === 'mainnet') {
+			this.constants = MAINNET_CONFIG;
+		}
+
+		if (network === 'testnet') {
+			this.constants = TESTNET_CONFIG;
+		}
+
+		if (config.packageIds) {
+			this.constants = { ...(this.constants || {}), ...config.packageIds };
+		}
+	}
+
+	/**
+	 * Returns the price list for SuiNS names.
+	 */
+	async getPriceList(): Promise<SuinsPriceList> {
+		if (!this.constants.suinsObjectId) throw new Error('Suins object ID is not set');
+		if (!this.constants.suinsPackageId) throw new Error('Price list config not found');
+
+		const priceList = await this.#client.getDynamicFieldObject({
+			parentId: this.constants.suinsObjectId,
+			name: {
+				type: getConfigType(
+					this.constants.suinsPackageId.v1,
+					getPricelistConfigType(this.constants.suinsPackageId.v1),
+				),
+				value: { dummy_field: false },
+			},
+		});
+
+		if (
+			!priceList ||
+			!priceList.data ||
+			!priceList.data.content ||
+			priceList.data.content.dataType !== 'moveObject' ||
```

Comment:
this can probably be simplified with optional chaining: `priceList?.data?.content?.dataType !== 'moveObject'`